### PR TITLE
Migrate to odo 1.0.0 release

### PR DESCRIPTION
**Pull request by dgolovin**

### CODE REVIEW

**Comment by jrichter1** [comment]
Also some bigger issues I didn't pinpoint in the code (list might not be complete):
 - not pushed component is only visible if its context folder is currently open in vscode workspace
 - pushed component moves to 'no context' state if its folder is not open in workspace
 - there is no way to delete components in 'not pushed' state
 - deleting application that has not been pushed (only exists locally) tries to remove it from the cluster, results in error "app does not exist"
 - deleting pushed components or applications does not remove their local context, so the reappear on the next refresh/reload
    - either delete both local and pushed parts, or have a separate 'undeploy' that removes them from the cluster only
 - creating component from workspace folder does nothing when no workspace folder is open
 - link component/service commands are broken due to changed odo syntax
 - creating a route requires the whole component to be pushed again :/
 - cancelling context folder selection results in 'TypeError: Cannot read property '0' of undefined'
 - creating component from binary file fails with '✗  this directory already contains a component' even though I've never used the given file/folder before
 - creating component from workspace folder in a new application results in it having 'no context' until workspace is reloaded

File: test/extension.test.ts [hunk 1]
```diff
@@ -22,23 +22,45 @@ import { Service } from '../src/openshift/service';
 import { Storage } from '../src/openshift/storage';
 import { Url } from '../src/openshift/url';
 import packagejson = require('../package.json');
+import path = require('path');
+import { OdoImpl, ContextType, OpenShiftObjectImpl } from '../src/odo';
+import { TestItem } from './openshift/testOSItem';
```
[the remark this answers is not in the document]
Comment (mohitsuman) [reply test/extension.test.ts]: fixed in https://github.com/redhat-developer/vscode-openshift-tools/pull/943

File: src/odo.ts [hunk 2]
```diff
@@ -399,49 +539,55 @@ export class OdoImpl implements Odo {
...
-            const stdout: string = result.stdout.trim();
-            if (stdout !== "" ) {
-                projs = stdout.split("\n").map<OpenShiftObject>((value) => new OpenShiftObjectImpl(cluster, value, ContextType.PROJECT, OdoImpl.instance));
+            let data: any[] = [];
+            try {
+                data = JSON.parse(result.stdout).items;
+            } catch (ignore) {
             }
-            return projs.sort(compareNodes);
+            const projs = data.map((value) => value.metadata.name);
```
[the remark this answers is not in the document]
Comment (mohitsuman) [reply src/odo.ts]: fixed in https://github.com/redhat-developer/vscode-openshift-tools/pull/943

**Comment by dgolovin** [comment]
@jrichter1 
> * not pushed component is only visible if its context folder is currently open in vscode workspace
This is not an issue, because extension is not supposed to be aware of all contexts in local filesystem.

> * pushed component moves to 'no context' state if its folder is not open in workspace
This is right behavior.

> * there is no way to delete components in 'not pushed' state
For now we stick to remove context folder from workspace, but there should be delete command for that #957.

> * deleting application that has not been pushed (only exists locally) tries to remove it from the cluster, results in error "app does not exist".
This should be fixed #958.

> * deleting pushed components or applications does not remove their local context, so they reappear on the next refresh/reload as not pushed
>   * either delete both local odo config and pushed parts, or have a separate 'undeploy' that removes them from the cluster only
I think we will stick to deleting both config and pushed parts with '--all' option

> * creating component from workspace folder does nothing when no workspace folder is open
That should be taken care of with recent fixes #923.

> * link component/service commands are broken due to changed odo syntax
Should be fixed #956.

> * creating a route requires the whole component to be pushed again :/
That is expected odo behavior. We should make devs aware of it in information message.

> * cancelling context folder selection results in 'TypeError: Cannot read property '0' of undefined'
Should be fixed #951.

> * creating component from binary file fails with '✗  this directory already contains a component' even though I've never used the given file/folder before
Should be fixed #947.

> * creating component from workspace folder in a new application results in it having 'no context' until workspace is reloaded
Should be fixed #959.

**Pull request merged**